File: jskat-swing-gui/src/main/java/org/jskat/gui/swing/table/BidBubblePanel.java
package org.jskat.gui.swing.table;

import org.jskat.gui.swing.LayoutFactory;

import javax.swing.*;
import java.awt.*;

/**
 * Panel for a bid bubble
 */
class BidBubblePanel extends JPanel {

    private static final long serialVersionUID = 1L;

    private final Image bubbleImage;
    private final JLabel bidLabel;

    BidBubblePanel(Image newBubbleImage) {

        bubbleImage = newBubbleImage;

        setLayout(LayoutFactory.getMigLayout("fill"));

        bidLabel = new JLabel();
        bidLabel.setFont(new Font(Font.DIALOG, Font.BOLD, 16));
        setBidValue(0);

        add(bidLabel, "center");
        setMinimumSize(new Dimension(75, 75));
    }

    void setBidValue(int bidValue) {

        if (bidValue > -1) {

            bidLabel.setText(String.valueOf(bidValue));

        } else {

            bidLabel.setText("X");
        }
    }

    @Override
    public void paintComponent(Graphics g) {

        g.drawImage(bubbleImage, 0, 0, null);
    }

}

File: jskat-gui/src/main/kotlin/org/jskat/gui/javafx/table/BidBubblePanel.kt
package org.jskat.gui.javafx.table

import javafx.application.Platform
import javafx.scene.control.Label
import javafx.scene.image.Image
import javafx.scene.image.ImageView
import javafx.scene.layout.StackPane
import javafx.scene.text.Font
import javafx.scene.text.FontWeight

class BidBubblePanel(bubbleImage: Image) : StackPane() {
    private val bidLabel = Label()

    init {
        val bubbleView = ImageView(bubbleImage)
        children.addAll(bubbleView, bidLabel)

        bidLabel.font = Font.font("Dialog", FontWeight.BOLD, 16.0)
        setBidValue(0)
    }

    fun setBidValue(bidValue: Int) {
        Platform.runLater {
            if (bidValue > -1) {
                bidLabel.text = bidValue.toString()
            } else {
                bidLabel.text = "X"
            }
        }
    }
}
